File: in_Java/0118 Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        if(numRows == 0) return res;
        List<Integer> t = new ArrayList<>();
        t.add(1);
        res.add(t);
        if(numRows == 1) return res;
        t = new ArrayList<>();
        t.add(1); t.add(1);
        res.add(t);
        if(numRows == 2) return res;
        for(int i = 3; i <= numRows; i++){
            t = new ArrayList<>();
            t.add(1);
            for(int id = 1; id < i-1; id++){
                t.add(res.get(i-2).get(id-1) + res.get(i-2).get(id));
            }
            t.add(1);
            res.add(t);
        }
        return res;
    }
}

File: in_Python/0118 Pascal's Triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        res = []
        if numRows == 0:
            return res
        res.append([1])
        if numRows == 1:
            return res
        res.append([1,1])
        if numRows == 2:
            return res
        for i in range(3, numRows+1):
            prev = res[-1]
            cur = []
            cur.append(1)
            for i in range(len(prev)-1):
                cur.append(prev[i] + prev[i+1])
            cur.append(1)
            res.append(cur)
        return res